### DISCUSSION TITLE
Ability to create bookmarks in documentation.

### DISCUSSION THREAD

**Discussion by UltraInstinct05** [squidfunk/mkdocs-material#4056] [Ideas]
This is just an idea I have, but it'd be kind of cool if mkdocs-material had a thing, where users can simply bookmark certain stuff, like sidebar links, TOC links etc. This can especially be useful for Guide books where a user can simply bookmark a certain section in sidebar or TOC and then get back to it later.

Not sure what shape this might take, but maybe like a bookmark icon next to a link that can be clicked to mark it as a bookmark (I believe bookmarks could be stored in local storage to make them persistent).

Anyways just an idea ! 🙂

**Comment by squidfunk** [comment]
There are some threads on [Hypothesis](https://web.hypothes.is/), which is a bookmark solution, but we decided against integration for several reasons. I invite you to search the issue tracker and link everything you discover here. For starters:

- #1145

**Reply by UltraInstinct05** [reply]
I took a look at Hypothesis and from what I understand it seems a much more sophisticated bookmarking solution which can bookmark any lines. The idea was more about a rather simplified solution. The sidebar + TOC pretty much gives a nice position of where a user might be in the documentation and I guess a solution to just to do a bookmark either a sidebar link or TOC was something I had in kind.

**Reply by squidfunk** [reply]
Could you explain how you would expect the mechanics to work from a user perspective? Ideally with all use cases you can imagine.

**Reply by UltraInstinct05** [reply]
What I had in mind was this.
- There would be an option in `mkdocs.yml` to turn, let's say a hypothetical `bookmark` option on.
- When such an option is turned on, then each link in sidebar and TOC would get an icon rendered next to them. This icon would have 2 states, empty and filled. Empty represents non bookmarked link, filled represents bookmarked state.
- If a user wants to bookmark a link (by link I mean a sidebar or TOC link), they just click the icon and it gets filled. If the website is closed, the bookmarks are restored when opened again.

Anyways, this is what I had in mind when I opened this idea.

**Reply by squidfunk** [reply]
The problem with this is that you can't rely on local storage to be persistent or even available. For example, in incognito mode, your local storage is not persisted. This makes it pretty much impossible to solve this locally without third-party services. For this reason, I'm not sure it makes sense, as the user could not rely on the links being persistently bookmarked.

**Reply by UltraInstinct05** [reply]
Ah, I was under the impression that the light/dark mode made use of local storage so it could be used. I guess without a 3rd party service, this might not be feasible, then. Thanks for considering !

**Reply by squidfunk** [reply]
The selected color scheme is stored in local storage, but it's an ephemeral value, meaning that when we lose local storage, the worst it can get is _wrong color scheme_, not _loss of data_ 😊 This is even less of an issue with the new [automatic mode](https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#automatic-light-dark-mode).
